WordPress/Classes/Utility/CoreDataHelper.swift

```diff
@@ -1,47 +1,44 @@
 import Foundation
 
 
-
-// MARK: - ManagedObject Base Class
+// MARK: - NSManagedObject Default entityName Helper
 //
-protocol ManagedObject {
-    static var entityName: String { get }
+extension NSManagedObject {
+    class var entityName: String {
+        return classNameWithoutNamespaces()
+    }
 }
 
 
-// MARK: - Core Data Helpers
+// MARK: - NSManagedObjectContext Helpers!
 //
-struct CoreDataHelper<T> where T: NSManagedObject, T: ManagedObject {
-    /// CoreData ManagedObjectContext
-    ///
-    let context: NSManagedObjectContext
-
+extension NSManagedObjectContext {
 
     /// Returns a new FetchRequest for the associated Entity
     ///
-    func newFetchRequest() -> NSFetchRequest<NSFetchRequestResult> {
-        return NSFetchRequest(entityName: T.entityName)
+    func newFetchRequest<T: NSManagedObject>(forType type: T.Type) -> NSFetchRequest<NSFetchRequestResult> {
```

Comment:
This probably shouldn't be in the extension of `NSManagedObjectContext` as it's not related to a context in any way until execution. `NSManagedObject` already has a `fetchRequest` class function on it for doing exactly this. I say delete and refactor as such: 

```swift
let request = T.fetchRequest()
```